README.md

```diff
@@ -156,6 +156,57 @@ We recommend that you always instantiate a client (e.g., with `client = OpenAI()
 - It's harder to mock for testing purposes
 - It's not possible to control cleanup of network connections
 
+## Integration Example  
+
+### FastAPI
+
+```py
+from openai import AsyncOpenAI
+from contextlib import asynccontextmanager
+from fastapi import Depends, FastAPI
+
+class AioClient:
+    """
+    async client with singleton approach
+    https://github.com/tiangolo/fastapi/discussions/8301
+    https://fastapi.tiangolo.com/advanced/events/
+    """
+    client: AsyncOpenAI | None = None
```

Comment:
It's important to note that the use of | as a Union operator is supported starting from Python 3.10, as detailed in [PEP 604](https://peps.python.org/pep-0604/). To ensure broader compatibility, especially with versions as early as Python 3.7, I recommend updating the code to use an alternative approach that aligns with the capabilities of these earlier Python versions